waku/v2/protocol/metadata/waku_metadata.go

```diff
@@ -209,6 +214,15 @@ func (wakuM *WakuMetadata) ListenClose(n network.Network, m multiaddr.Multiaddr)
 	// Do nothing
 }
 
+func (wakuM *WakuMetadata) disconnectPeer(peerID peer.ID, reason error) {
+	logger := wakuM.log.With(logging.HostID("peerID", peerID))
+	logger.Error("disconnecting from peer", zap.Error(reason))
+	wakuM.h.Peerstore().RemovePeer(peerID)
+	if err := wakuM.h.Network().ClosePeer(peerID); err != nil {
+		logger.Error("could not disconnect from peer", zap.Error(err))
```

Comment:
This error is interesting, wondering what could be the reason for this failure.
If it is just that remote-peer has disconnected before us, then maybe we shouldn't log error.
If not, would adding a retry here make it more fool-proof?